### PR TITLE
Rewrite everything

### PR DESCRIPTION
**Pull request by Mcbeany**
## So, what happened to **BetterMinion**

This plugin is expected to be released in March of this year as well as PocketMine 3's EOL. however we are still facing some problems in developing it (specifically we don't know what to do next xD). Besides, I'm quite busy at the moment and want to spend more time studying. We will keep the source code of PM4 and PM3 versions until we finalize this rewritten version.

Release date may be moved to the summer, sorry!

### CODE REVIEW

**Comment by Mcbeany** [comment]
wtf is this type of message :joy:

File: src/minions/entities/BaseMinion.php [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+			->setTag(MinionNBT::INFORMATION, $this->minionInformation->serializeTag())
+			->setTag(MinionNBT::INVENTORY, $this->minionInventory->serializeTag());
+	}
+
+	/**
+	 * Do check thing before start working.
+	 */
+	public function onUpdate(int $currentTick) : bool{
+		if($this->isWorking()){
+			$lastItem = $this->minionInventory->getItem($this->minionInventory->getSize() - 1);
```
Comment (NgLamVN) [inline src/minions/entities/BaseMinion.php:77]: This check should be put at another function in MinionInventory like MinionInventory::isFull().

File: src/minions/entities/BaseMinion.php [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+				}
+				$this->onAction();
+			}else{
+				$this->doOfflineAction($times);
+			}
+		}
+		return parent::entityBaseTick($tickDiff);
+	}
+
+	public function setNameTag(string $name = "") : void{
```
Comment (NgLamVN) [inline src/minions/entities/BaseMinion.php:115]: $name can be null for empty check, "" still can be a nametag :>

File: src/minions/entities/BaseMinion.php [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+	 * Returns original name of the minion, include its owner.
+	 */
+	public function getOriginalNameTag() : string{
+		return $this->ownerName . "'s Minion";
+	}
+
+	/**
+	 * Returns the minion's working radius.
+	 */
+	public function getWorkingRadius() : int{
```
Comment (NgLamVN) [inline src/minions/entities/BaseMinion.php:257]: Working Radius should put at default value and can be overwrite in other type of the Minion

**Review by NgLamVN — CHANGES_REQUESTED** [review]

**Comment by Mcbeany** [comment]
What's next?
Agh! Farming minion & menu

**Pull request merged**